addOns/retire/src/main/java/org/zaproxy/addon/retire/model/Repo.java

```diff
@@ -128,6 +131,19 @@ public Result scanJS(HttpMessage msg) {
         return scanHash(hash);
     }
 
+    private String getCleanContent(HttpMessage msg) {
+        if (msg.getResponseHeader().isHtml()) {
+            Source source = new Source(msg.getResponseBody().toString());
```

Comment:
Better use existing `Source`.

Reply:
The method is "public" can I still change the signature? (Just do a full version bump in the add-on?)

Reply:
Who's calling this class directly? If you really want to keep binary compatibility leave the old method (which creates the `Source`) calling the new one.

Reply:
I doubt anyone else is, but I was trying to do things "right".

Reply:
This is more an internal class though, which should not be part of the public API.